File: 2019/day01/part1.js
const fs = require('fs');

fs.readFile('input.txt', { encoding: 'utf-8' }, (err, data) => {
    if (err) throw err;
    const arr = data.split("\n");
    let sum = 0;
    for (let i = 0; i < arr.length; i++) {
        sum += Math.floor(arr[i] / 3) - 2;
    }
    console.log(sum)
});

// or

fs.readFile('input.txt', { encoding: 'utf-8' }, (err, data) => {
    if (err) throw err;
    const arr = data.split("\n");
    console.log(arr.reduce((a, b) => Math.floor(b / 3) - 2 + a, 0))
});

// 94794

console.log(Math.floor(94794 / 3) - 2)

File: 2019/day01/part1.py
import math

with open('input.txt') as fp:
    line = fp.readline()
    sum = 0
    while line:
        # "//" is a floor division apparently :~) that's pretty cool
        sum += int(line) // 3 - 2
        # sum += math.floor(int(line) / 3) - 2
        line = fp.readline()
    print(sum)
